src/WsseMiddleware.php

```diff
@@ -7,79 +7,48 @@
 use Psr\Http\Message\RequestInterface;
 use Psr\Http\Message\ResponseInterface;
 use RobRichards\WsePhp\WSSESoap;
-use RobRichards\XMLSecLibs\XMLSecurityKey;
 use Soap\Psr18Transport\Xml\XmlMessageManipulator;
+use Soap\Psr18WsseMiddleware\WSSecurity\Entry\WsseEntry;
 use VeeWee\Xml\Dom\Document;
 
 final class WsseMiddleware implements Plugin
 {
-    private string $privateKeyFile;
-    private string $publicKeyFile;
-    private string $serverCertificateFile = '';
-    private int $timestamp = 3600;
-    private bool $signAllHeaders = false;
-    private string $digitalSignMethod = XMLSecurityKey::RSA_SHA1;
-    private string $userTokenName = '';
-    private string $userTokenPassword = '';
-    private bool $userTokenDigest = false;
-    private bool $encrypt = false;
-    private bool $hasUserToken = false;
-    private bool $serverCertificateHasSubjectKeyIdentifier = true;
-
-    public function __construct(string $privateKeyFile, string $publicKeyFile)
-    {
-        $this->privateKeyFile = $privateKeyFile;
-        $this->publicKeyFile = $publicKeyFile;
-    }
-
-    public function withTimestamp(int $timestamp = 3600): self
-    {
-        $new = clone $this;
-        $new->timestamp = $timestamp;
-
-        return $new;
-    }
-
-    public function withAllHeadersSigned(): self
-    {
-        $new = clone $this;
-        $new->signAllHeaders = true;
-
-        return $new;
-    }
-
-    public function withDigitalSignMethod(string $digitalSignMethod): self
-    {
-        $new = clone $this;
-        $new->digitalSignMethod = $digitalSignMethod;
-
-        return $new;
+    /**
+     * @var list<WsseEntry>
+     */
+    private array $outgoingEntries;
+    /**
+     * @var list<WsseEntry>
+     */
+    private array $incomingEntries;
+    private bool $mustUnderstand = false;
+    private ?string $actor = null;
+
+    /**
+     * @no-named-arguments
+     * @param list<WsseEntry> $outgoing
+     * @param list<WsseEntry> $incoming
+     */
+    public function __construct(
+        array $outgoing = [],
```

Comment:
Alternatively, an OutgoingEntry or IncomingEntry interface could be wrapped arround a WsseEntry.
Not sure what the best way is yet.

By explicitely specifying it in the constructor, you have full control and there is no magic involved.
However, it requires you to know more about WSSE. Maybe that's not a bad thing...